### PR TITLE
OSASINFRA-3626: Resize support with the OpenStack Manila CSI Driver

### PR DESCRIPTION
**Pull request by gouthampacha**
Volume expansion is a mandatory feature supported
within back end drivers in OpenStack Manila. To
enable this with the Manila CSI driver, the operator needs to specify the external resizer sidecar along with its appropriate RBAC, and set "allowVolumeExpansion" to True. Volume expansion for NAS shared file systems is a controller-only operation; there's no node plugin operations necessary unlike block storage.

### CODE REVIEW

**Comment by gouthampacha** [comment]
/retest

Manila e2e tests didn't execute.. cluster creation failed

**Review by mandre — COMMENTED** [review]
The expansion tests that were previously skipped are now running, this is good. However, they're all failing at the moment. Looking at [the csi-resizer logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_csi-operator/322/pull-ci-openshift-csi-operator-master-e2e-openstack-manila-csi/1859754809997922304/artifacts/e2e-openstack-manila-csi/gather-extra/artifacts/pods/openshift-manila-csi-driver_openstack-manila-csi-controllerplugin-dc66b86c4-6bdlx_csi-resizer.log):
```
I1122 02:46:14.370331       1 event.go:389] "Event occurred" object="e2e-volume-expand-6734/manila.csi.openstack.orgvnwnh" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Normal" reason="Resizing" message="External resizer is resizing volume pvc-3393a818-fac2-44ed-8444-6f2feb036660"
E1122 02:46:14.374209       1 controller.go:314] "Error syncing PVC" err="resize volume \"pvc-3393a818-fac2-44ed-8444-6f2feb036660\" by resizer \"manila.csi.openstack.org\" failed: rpc error: code = InvalidArgument desc = volume expand secrets cannot be nil or empty"
I1122 02:46:14.375213       1 event.go:389] "Event occurred" object="e2e-volume-expand-6734/manila.csi.openstack.orgvnwnh" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Warning" reason="VolumeResizeFailed" message="resize volume \"pvc-3393a818-fac2-44ed-8444-6f2feb036660\" by resizer \"manila.csi.openstack.org\" failed: rpc error: code = InvalidArgument desc = volume expand secrets cannot be nil or empty"
```

Are we missing a volume mount or something?

**Comment by mandre** [comment]
You'll need to run `make update` too.

File: pkg/driver/openstack-manila/openstack_manila.go [hunk 1]
```diff
@@ -72,7 +72,10 @@ func GetOpenStackManilaGeneratorConfig() *generator.CSIDriverGeneratorConfig {
 					"--timeout=120s",
 					"--feature-gates=Topology=true",
 				),
-				commongenerator.DefaultResizer.WithExtraArguments(),
+				commongenerator.DefaultResizer.WithExtraArguments(
+					"--timeout=240s",
+					"-handle-volume-inuse-error=false",
```
[the remark this answers is not in the document]
Comment (mandre) [reply pkg/driver/openstack-manila/openstack_manila.go:77]: We'll need to set the `csi.storage.k8s.io/controller-expand-secret-name` and `csi.storage.k8s.io/controller-expand-secret-namespace` params in the storage class.

https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html

**Comment by mandre** [comment]
@gouthampacha As we're a bit pressed by time, I took the liberty to amend your commit with fixes and pushed a new PR at https://github.com/openshift/csi-operator/pull/327 so that we can confirm CI is happy.

File: pkg/driver/openstack-manila/openstack_manila.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gouthampacha) [reply pkg/driver/openstack-manila/openstack_manila.go]: > We'll need to set the csi.storage.k8s.io/controller-expand-secret-name and csi.storage.k8s.io/controller-expand-secret-namespace params in the storage class.

Thank you! I missed this

**Review by mandre — COMMENTED** [review]
The volume expansion tests are now working. The [failure seen in the latest run](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_csi-operator/322/pull-ci-openshift-csi-operator-master-e2e-openstack-manila-csi/1859901510339530752) is unrelated to the PR, as shown by [another CI run](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_csi-operator/327/pull-ci-openshift-csi-operator-master-e2e-openstack-manila-csi/1859879919933198336) performed as part of [another PR](https://github.com/openshift/csi-operator/pull/327) that contained the exact same code.
/lgtm
/approve

**Comment by mandre** [comment]
/label px-approved
@rlobillo for the qe-approved label

**Comment by gouthampacha** [comment]
/retest-required

failures continue to be unrelated CI flakiness..

**Comment by mandre** [comment]
The change is limited to manila... let's get this merged.
/override ci/prow/hypershift-aws-e2e-external

**Pull request merged**